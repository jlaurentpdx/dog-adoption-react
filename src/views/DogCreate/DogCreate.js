import { useState } from 'react';
import { useHistory } from 'react-router';
import { createDog } from '../../services/dogs';
import DisplayForm from '../../components/DisplayForm/DisplayForm';

export default function DogCreate() {
  const welcome = 'New entry';
  const submitText = 'Add';
  const history = useHistory();
  const [dog, setDog] = useState({});
  const [message, setMessage] = useState('');

  const upDog = (key, value) => {
    dog[key] = value;
    setDog({ ...dog });
  };

  const handleSubmit = async (e) => {
    try {
      e.preventDefault();
      await createDog(dog);
      setMessage('new dog added! redirecting...');
      setTimeout(() => {
        history.goBack();
      }, 2000);
    } catch {
      setMessage('something went wrong');
    }
  };

  return (
    <DisplayForm
      welcome={welcome}
      message={message}
      upDog={upDog}
      submitText={submitText}
      handleSubmit={handleSubmit}
    />
  );
}
