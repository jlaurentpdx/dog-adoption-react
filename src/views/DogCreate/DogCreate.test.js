import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import DogCreate from './DogCreate';

test('DogCreate should render an empty form', () => {
  const { container } = render(
    <MemoryRouter>
      <DogCreate />
    </MemoryRouter>
  );

  screen.getAllByDisplayValue('');

  expect(container).toMatchSnapshot();
});
