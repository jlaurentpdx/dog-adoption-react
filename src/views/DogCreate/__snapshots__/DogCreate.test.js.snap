// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogCreate should render an empty form 1`] = `
<div>
  <div>
    <p>
      This is the edit page.
    </p>
    <p />
    <form>
      <div
        class="form-control"
      >
        <label>
          Name: 
        </label>
        <input
          placeholder="enter name"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-control"
      >
        <label>
          Description: 
        </label>
        <textarea
          placeholder="Who is this dog?"
          type="text"
        />
      </div>
      <div
        class="form-control"
      >
        <label>
          Age: 
        </label>
        <input
          placeholder="enter age in years"
          type="number"
          value=""
        />
      </div>
      <div
        class="form-control"
      >
        <label>
          Image: 
        </label>
        <input
          placeholder="enter URL"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-control"
      >
        <label>
          Breed: 
        </label>
        <input
          placeholder="enter breed or mix"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-control"
      >
        <button>
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;
